lib/matplotlib/axes/_base.py

```diff
@@ -3195,7 +3200,7 @@ def get_children(self):
         children.extend(self.lines)
         children.extend(self.texts)
         children.extend(self.artists)
-        children.extend(six.itervalues(self.spines))
+        children.extend(_sorted_itervalues(self.spines))
```

Comment:
Now that the test case is implemented it looks like this is the only change that is required in this file.